cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -521,8 +529,14 @@ public void run() {
         Iterable<String> sparkKeys = Iterables.filter(System.getProperties().stringPropertyNames(),
                                                       Predicates.containsPattern("^spark\\."));
         for (String key : sparkKeys) {
-          LOG.debug("Removing Spark system property: {}", key);
```

Comment:
Because it keep introducing new system properties that will keep increasing the memory usage. It is needed for SDK.